### PR TITLE
Fixes #99 - Fixed tests related to package-locking

### LINKED ISSUE #99: Fix tests related to package-locking and add new test scenarios as well.
Refs FM PR: [https://github.com/theforeman/foreman_maintain/pull/278](https://github.com/theforeman/foreman_maintain/pull/278)

### PR DESCRIPTION
**Pull request by jameerpathan111**

### CODE REVIEW

File: tests/test_packages.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
 from testfm.decorators import run_only_on
 from testfm.packages import Packages
 from testfm.log import logger
+from testfm.constants import SAT_HOSTNAME
+from fabric import Connection
```
Comment (ntkathole) [inline tests/test_packages.py]: sort in order

File: tests/test_packages.py [hunk 2]
```diff
@@ -115,17 +169,37 @@ def test_positive_lock_package_versions(setup_install_pkgs, ansible_module):
...
-    for result in contacted.values():
-        logger.info(result['stdout'])
-        assert 'satellite' in result['stdout']
+    contacted = ansible_module.yum(
+        name='zsh',
+        state='absent'
+    )
+    for result in contacted.values():
+        assert result['rc'] == 0
+    contacted = ansible_module.command('yum install -y zsh')
```
Comment (ntkathole) [inline tests/test_packages.py]: why not ansible_module.yum()

File: tests/test_packages.py [hunk 3]
```diff
@@ -115,17 +169,37 @@ def test_positive_lock_package_versions(setup_install_pkgs, ansible_module):
...
+    for result in contacted.values():
+        assert result['rc'] == 1
+        assert "Use foreman-maintain packages install/update <package>" in result['stdout']
+    # Test whether FM packages install/ update command works as expected.
+    contacted = Connection(SAT_HOSTNAME, 'root').run(
```
Comment (ntkathole) [inline tests/test_packages.py]: Create a common function where I will import it and use it
```
ex: 
from common_function import run

run(Packages.install())
```

File: tests/test_packages.py [hunk 4]
```diff
@@ -136,20 +210,23 @@ def test_positive_lock_package_versions(setup_install_pkgs, ansible_module):
...
+    contacted = ansible_module.yum(
+        name='zsh',
+        state='absent'
+    )
     for result in contacted.values():
-        logger.info(result['stdout'])
-        assert 'satellite' not in result['stdout']
-    contacted = ansible_module.shell("yum versionlock list | wc -l")
+        assert result['rc'] == 0
+    contacted = ansible_module.command('yum install -y zsh')
```
Comment (ntkathole) [inline tests/test_packages.py]: same as above

File: tests/test_packages.py (hunk 2 above)
Comment (jameerpathan111) [reply tests/test_packages.py]: Because I want to check output of yum command. It should give message about using FM to install package.

Comment (ntkathole) [reply tests/test_packages.py]: @jameerpathan111 can you get it from ansible command stdout?

Comment (jameerpathan111) [reply tests/test_packages.py]: no, that's why I am using command module instead of yum.

File: tests/test_packages.py (hunk 1 above)
Comment (jameerpathan111) [reply tests/test_packages.py]: updated PR

File: tests/test_packages.py (hunk 3 above)
Comment (jameerpathan111) [reply tests/test_packages.py]: Added it in testfm.helpers

**Comment by ntkathole** [comment]
ok to test : tests/test_packages.py

**Comment by jameerpathan111** [comment]
Refs: 
- [https://github.com/SatelliteQE/robottelo-ci/pull/1606](https://github.com/SatelliteQE/robottelo-ci/pull/1606)
- [https://github.com/SatelliteQE/robottelo-ci/pull/1600](https://github.com/SatelliteQE/robottelo-ci/pull/1600)

**Comment by jameerpathan111** [comment]
Test Results:

```
(env-testfm) [jpathan@jpathan testfm]$ pytest -s --ansible-host-pattern satellite --ansible-user root --ansible-inventory testfm/inventory tests/test_packages.py 
=================================================================================== test session starts ===================================================================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.8.0, pluggy-0.6.0
ansible: 2.6.14
rootdir: /home/jpathan/projects/testfm, inifile:
plugins: ansible-2.0.1
collected 3 items                                                                                                                                                                         
********  output omitted  ***********
==================================================================================== warnings summary =====================================================================================
tests/test_packages.py::test_positive_fm_packages_install
  /home/jpathan/projects/env-testfm/lib64/python3.6/site-packages/paramiko/kex_ecdh_nist.py:39: CryptographyDeprecationWarning: encode_point has been deprecated on EllipticCurvePublicNumbers and will be removed in a future version. Please use EllipticCurvePublicKey.public_bytes to obtain both compressed and uncompressed point encoding.
    m.add_string(self.Q_C.public_numbers().encode_point())
  /home/jpathan/projects/env-testfm/lib64/python3.6/site-packages/paramiko/kex_ecdh_nist.py:96: CryptographyDeprecationWarning: Support for unsafe construction of public numbers from encoded data will be removed in a future version. Please use EllipticCurvePublicKey.from_encoded_point
    self.curve, Q_S_bytes
  /home/jpathan/projects/env-testfm/lib64/python3.6/site-packages/paramiko/kex_ecdh_nist.py:111: CryptographyDeprecationWarning: encode_point has been deprecated on EllipticCurvePublicNumbers and will be removed in a future version. Please use EllipticCurvePublicKey.public_bytes to obtain both compressed and uncompressed point encoding.
    hm.add_string(self.Q_C.public_numbers().encode_point())

-- Docs: http://doc.pytest.org/en/latest/warnings.html
========================================================================= 3 passed, 3 warnings in 1876.12 seconds =========================================================================

```

**Pull request merged**